### ISSUE TITLE
Only using one (1) CPU core

### ISSUE THREAD

**Issue by TheDailySpank** [s3inlc/hashtopolis#540]
- Your current Server version located at the bottom of any Hashtopolis webpage: 0.10.1+repository
- Current Client version: "up to date"
- Your current Hashcat version: v5.1.0-42-g471a8ccc (OS X) / v5.1.0 (Ubuntu)
- The exact task command you are trying to run. Any work done using Hashcat
- Debug output from the client by running "hashtopolis.exe -d" or with debug flag set on the python client: IDK how to do that, I'm not so good with python, sorry.

When running the python client, everything works as expected, files get downloaded, hashcat runs, returns results, status are shown in hashtopolis. However, I noticed my local machine's hashing speed was much lower than expected while running the same task manually, using the hashcat executable from inside the "crackers/1" folder it downloaded.

I have tried on two separate Ubuntu servers with the OpenCL CPU only versions running as well as my local machine (Mac) set as CPU Only: Yes and No.

All three systems show multiple instances of hashcat running that matches but only one is fully utilizing the CPU.

By pulling up a virtual machine on my local machine, while running the hashtopolis client locally, I am able to see an increase in the hashrate which to me proves it's not directly an OpenCL issue.

Is there a way to show the exact command the hashtopolis client is using or is there a configuration option I am not aware of somewhere I can switch?

**Comment by winxp5421** [comment]
Please see https://github.com/s3inlc/hashtopolis-agent-python for debug options

**Comment by TheDailySpank** [comment]
I failed to mention it's a prince attack. Would that, by virtue of it being a stream be the limiting factor?

Using the debug flag I was able to get the command that is being run by hashtopolis. I tried that manually and got the same speed with only one CPU being fully utilized. I removed the prince processor part and re-ran and got the expected speeds with all cores being utilized.

Solution: Add rules when running prince attacks so as to not deplete the output from pp64 when attacking fast hashes such as SHA1.
